packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -452,31 +440,17 @@ function getType(
 		possibleTypes.length !== 0,
 		0x84e /* data is incompatible with all types allowed by the schema */,
 	);
-	checkInput(
-		possibleTypes.length === 1,
-		() =>
+	if (possibleTypes.length !== 1) {
```

Comment:
checkInput is only used here, and I don't think it improved the readability. I considered keeping it but it was easier to inline it here than to come up with a good doc comment for it.

With this change there is less code, less module level methods, and simpler flow control. The docs on invalidInput seem redundent with using UsageError, so I think just using it directly accomplishes the same clarity without needing extra docs, and better matches the rest of the code in FF (and elsewhere in this file)